wafer/schedule/models.py

```diff
@@ -261,6 +281,25 @@ def invalidate_check_schedule(*args, **kw):
     check_schedule.invalidate()
 
 
+def update_schedule_items(*args, **kw):
+    """We save all the schedule items associated with this slot, so
+       the last_update time is updated to reflect any changes to the
+       timing of the slots"""
+    slot = kw.pop('instance', None)
+    if not slot:
+        return
+    for item in slot.scheduleitem_set.all():
+       item.save()
+    # We also need to update the next slot, in case we changed it's
+    # times as well
+    next_slot = slot.slot_set.all()
+    if next_slot.count():
+        # From the way we structure the slot tree, we know that
+        # there's only 1 next slot that could have changed.
+        for item in next_slot[0].scheduleitem_set.all():
+            item.save()
```

Comment:
Are we sure that we can't trigger a save loop if two slots form a cycle?

Reply:
The validation on slots should ensure we can't have a slot cycle, but restricting these to only save the update_date is probably a good idea.